src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,125 @@
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t create manufacturer. "
+                    + manufacturer + " ", e);
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        String getManufacturerRequest
+                = "SELECT * FROM manufacturers WHERE id = ? AND is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getStatement
+                        = connection.prepareStatement(getManufacturerRequest)) {
+            getStatement.setLong(1, id);
+            ResultSet resultSet = getStatement.executeQuery();
+            Manufacturer manufacturer = null;
+            if (resultSet.next()) {
+                manufacturer = getManufacturer(resultSet);
+            }
+            return Optional.ofNullable(manufacturer);
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Can't get manufacturer by id "
+                    + id + " ", throwable);
+        }
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        String getAllRequest = "SELECT * FROM manufacturers WHERE is_deleted = false;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllStatement = connection.prepareStatement(getAllRequest)) {
+            List<Manufacturer> manufacturers = new ArrayList<>();
+            ResultSet resultSet = getAllStatement.executeQuery(getAllRequest);
+            while (resultSet.next()) {
+                manufacturers.add(getManufacturer(resultSet));
+            }
+            return manufacturers;
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Can`t get manufacturers from table. ", throwable);
+        }
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        String updateManufacturersRequest
+                = "UPDATE manufacturers SET name = ?, country = ? WHERE id = ?"
+                + " AND is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateManufacturersStatement
+                        = connection.prepareStatement(updateManufacturersRequest,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            updateManufacturersStatement.setString(1, manufacturer.getName());
+            updateManufacturersStatement.setString(2, manufacturer.getCountry());
+            updateManufacturersStatement.setLong(3, manufacturer.getId());
+            updateManufacturersStatement.executeUpdate();
+            ResultSet generatedKeys = updateManufacturersStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                manufacturer.setId(id);
+            }
```

Comment:
id isn't being changed in this metho
```suggestion
        try (Connection connection = ConnectionUtil.getConnection();
                PreparedStatement updateManufacturersStatement
                        = connection.prepareStatement(updateManufacturersRequest) {
            updateManufacturersStatement.setString(1, manufacturer.getName());
            updateManufacturersStatement.setString(2, manufacturer.getCountry());
            updateManufacturersStatement.setLong(3, manufacturer.getId());
            updateManufacturersStatement.executeUpdate();             
```